website/source/docs/guides/production-acls.html.md

```diff
@@ -0,0 +1,484 @@
+agents. You will need to include the policy in the `consul acl token create`
+command.
+
+``` 
+$ consul acl token create -description "consul-server-one agent token" -policy-name consul-server-one 
+``` 
+
+This command returns the token information, which should include a description
+and policy information.
+
+Repeat this process for each agent. It is the responsibility of the operator to
+save tokens in a secure location; we recommend
+[Vault](https://www.vaultproject.io/).  
+
+### Add the token to the Agent.
+
+Finally, apply the tokens to the agents by adding the token to the agent
+configuration file and restarting the Consul service. Start with the servers
+and ensure they are working correctly before applying the client tokens. Below
+is an example server configuration file with an agent token.
+
+````
+# consul-server-one.hcl
+{ 
+"acl": { 
+  "enabled": true, 
+  "default_policy": "deny", 
+  "tokens": { 
+  	"agent": "e67a4a33-6579-b766-ac97-a0408fde3957" } 
+  }
+} 
+```` 
+
+At this point, every agent that has a token and has been restarted can once
+again read and write information to Consul, but only for node-related actions.
+Actions for individual services are not yet allowed.
+
+~> Note: If you are bootstrapping ACLs on and existing datacenter, remember to
+update the default policy to `default_policy = deny` and initiate another
+rolling restart. After applying the token. 
+
+#### Advantages of the ACL HTTP API
+
+Alternatively, you can apply the agent token with the [HTTP
+API](https://www.consul.io/api/acl/tokens.html#create-a-token). The HTTP API
+has three advantages.
+
+1. The token can be applied immediately to the agent, no service restart required.
+2. The configuration file does not need to be manually updated.  The token will
+not be persisted to disk, it will not be stored in plain text. 
+3. The final advantage can also be a disadvantage. Since the token is not
+persisted to disk, any time the agent is restarted the token will need to
+re-applied. This may not be an issue for scheduled maintenance, but could
+complicate an unplanned server outage.  
+
+Now that the agents have tokens, we can
+create tokens for the services.  
+
+## Apply Individual Tokens to the Services
+
+The token creation and application process for services is similar to agents. 
+Create a policy.  Use that policy to create a token.  Add the token to the
+service configuration file. 
+
+Review the [service
+rules](https://www.consul.io/docs/agent/acl-rules.html#service-rules) before
+getting started.
+
+Below is an example service definition that needs a token after bootstrapping
+the ACL system.
+
+``` 
+{ 
+  "service": { 
+    "name": "dashboard", 
+    "port": 9002, 
+    "check": { 
+    	"id": "dashboard-check", 
+    	"http": "http://localhost:9002/health", 
+    	"method": "GET",
+    	"interval": "1s", 
+    	"timeout": "1s" 
+    	} 
+    } 
+} 
+```
+
+This service definition should be located in the [configuration
```

Comment:
Alternatively services registered via the API will use the token present in the X-Consul-Token/Authorization HTTP header (or falling back to an agents default token). So they don't have to be put in the config file.

Not sure how we want to communicate this. It is not unexpected to have something register a service via the API when the HTTP port is localhost only.

Reply:
I believe this is addressed below. Re-open this comment if not.